File: exercises/practice/largest-series-product/largest_series_product.zig
pub const SeriesError = error{
    InvalidCharacter,
    NegativeSpan,
    InsufficientDigits,
};

pub fn largestProduct(digits: []const u8, span: i32) SeriesError!u64 {
    _ = digits;
    _ = span;
    @compileError("please implement the largestProduct function");
}

File: generators/exercises/largest_series_product.py
from lib import zstr, is_error


def gen_case(case):
    inp = case["input"]
    digits = inp["digits"]
    span = inp["span"]
    e = case["expected"]

    if is_error(e):
        if span < 0:
            err = "NegativeSpan"
        elif span > len(digits):
            err = "InsufficientDigits"
        else:
            err = "InvalidCharacter"
        return (
            f"    const actual = largest_series_product.largestProduct({zstr(digits)}, {span});\n"
            f"    try testing.expectError(largest_series_product.SeriesError.{err}, actual);\n"
        )

    return (
        f"    const expected: u64 = {e};\n"
        f"    const actual = try largest_series_product.largestProduct({zstr(digits)}, {span});\n"
        f"    try testing.expectEqual(expected, actual);\n"
    )
